src/deck/guides/3.0-upgrade.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Upgrade to Kong Gateway 3.x
+content-type: reference
+---
+
+With the [release of {{site.base_gateway}} 3.0](/gateway/changelog/), paths that use regular expressions are no longer automatically detected.
+For {{site.base_gateway}} to parse a path that contains a regular expression, the path must be prefixed with a **`~`**.
+
+For example, in 2.x, a route that was expressed like this:
+
+```
+/status/\d+
+```
+
+Would need to be rewritten in the new 3.0 format:
+
+```
+~/status/\d+
+```
+
+If you [migrated your {{site.base_gateway}} database](/gateway/latest/upgrade/) from 2.8.x to 3.0, the route paths in the database will have had the `~` prefix added automatically.
+This causes configuration drift between the formats of the routes that exist in the database and the routes in your configuration file.
+
+Before using your state files to update the database, convert them into the 3.0 format using `deck convert`.
+
+{:.important}
+> **Important**: Don't use `deck sync` with {{site.base_gateway}} 3.x before converting paths into the 3.0 format.
+This will break all regex routing in 3.x.
+
+## Convert declarative configuration files
+
+1. Run `deck-convert` against your 2.x state file to turn it into a 3.x file:
+
+    ```sh
+    deck convert \
+    --from kong-gateway-2.x \
+    --to kong-gateway-3.x \
+    --input-file kong.yaml \
+    --output-file new-kong.yaml
+    ```
+
+    This command upgrades the `_format_version` setting to `3.0` and adds the `~` prefix to all route paths containing a regex pattern.
+
+    You can leave `input-file` out to convert all declarative configuration files in your working directory.
+    We recommend specifying both input and output files.
+
+2. Validate the converted files in a test environment.
+
+    Perform manual audits of the generated file before applying the configuration in production.
```

Comment:
Does this mean running a deck diff? Or some combination of `deck validate`, `deck diff`? If so we should express that explicitly in step 2. If it's more complicated than that, then we should leave it.

Reply:
It's more complicated than that. As in, you have to literally look at your file and make sure that everything is correct.

Reply:
Rephrased for clarity.